### PR TITLE
Add Typescript as a language

### PR DESCRIPTION
**Pull request by balanza**
I think it's worth the effort to be clear on WHY we choose to work in Typescript

- [x] I've added a short description of the entry that
      clarifies what it does and why it's useful
- [x] I've added links to the entry's documentation and/or
      website
- [x] My frontmatter is correct, complete and follows the
      format specified in the [README](../README.md)
- [x] In case the entry is in the `adopt` category: I've
      added at least one reference to showcase the entry usage
      among our organization
- [x] I've described at least one use case for the entry

### CODE REVIEW

**Comment by gunzip** [comment]
Lgtm. I'd wait for some other CODEOWNER to review and approve this one.

File: radar/languages-and-frameworks/typescript.md
Comment (lucacavallaro) [inline radar/languages-and-frameworks/typescript.md:2]:
```suggestion
title: "TypeScript"
```

Comment (lucacavallaro) [reply radar/languages-and-frameworks/typescript.md:2]: image

File: radar/languages-and-frameworks/typescript.md
Comment (lucacavallaro) [inline radar/languages-and-frameworks/typescript.md:13]: type annotations also help increase DX because they allow IDEs to have better code autocompletion

File: radar/languages-and-frameworks/typescript.md
Comment (lucacavallaro) [inline radar/languages-and-frameworks/typescript.md:20]: these imho are the use cases of JavaScript, I would just say that ts can be used wherever js is used

**Review by lucacavallaro — APPROVED** [review]
LGTM ✅ 

I just left small notes

File: radar/languages-and-frameworks/typescript.md
Comment (ZorbaDimatteo) [inline radar/languages-and-frameworks/typescript.md:8]: @balanza probably there is a typo "..meaning that an application written in Typescript will be executed in **a** Javascript runtime"

Comment (balanza) [reply radar/languages-and-frameworks/typescript.md:8]: What's the typo? I don't see it 🙃

File: radar/languages-and-frameworks/typescript.md
Comment (ZorbaDimatteo) [inline radar/languages-and-frameworks/typescript.md:17]: also here "transpiled" instead of "transpilled"

**Review by ZorbaDimatteo — APPROVED** [review]

File: radar/languages-and-frameworks/typescript.md
Comment (balanza) [reply radar/languages-and-frameworks/typescript.md:13]: So we can add a page for type annotations 😉

File: radar/languages-and-frameworks/typescript.md
Comment (ZorbaDimatteo) [reply radar/languages-and-frameworks/typescript.md:8]: sorry! it's a finesse
"..will be executed in **the a** Javascript runtime" should be "..will be executed in **a**Javascript runtime" 
if I intended your sentence correctly

File: radar/languages-and-frameworks/typescript.md
Comment (balanza) [reply radar/languages-and-frameworks/typescript.md:20]: Not really. As per our history, we prefer Typescript for such cases. 

Should we introduce the use of Javascript instead/alongside?  Let's discuss it in https://github.com/pagopa/technology-radar/discussions/29

File: radar/languages-and-frameworks/typescript.md
Comment (balanza) [reply radar/languages-and-frameworks/typescript.md:8]: Great catch, I really didn't see it thanks

File: radar/languages-and-frameworks/typescript.md
Comment (balanza) [reply radar/languages-and-frameworks/typescript.md]: Fixed in https://github.com/pagopa/technology-radar/pull/28/commits/68ef2cb3e5dce18bb1a6c3de80ac0c1a0c34c645

File: radar/languages-and-frameworks/typescript.md
Comment (balanza) [reply radar/languages-and-frameworks/typescript.md]: Fixed in https://github.com/pagopa/technology-radar/pull/28/commits/68ef2cb3e5dce18bb1a6c3de80ac0c1a0c34c645

File: radar/languages-and-frameworks/typescript.md
Comment (lucacavallaro) [reply radar/languages-and-frameworks/typescript.md:13]: imho type annotation are an indistinguishable part of typescript as a language, I wouldn't put them as a separate entry

File: radar/languages-and-frameworks/typescript.md
Comment (lucacavallaro) [reply radar/languages-and-frameworks/typescript.md:20]: you're right, but honestly I've never had to choose whether to use typescript. I've always perceived it as our default language (in JS environments)

to me it's like `ESlint`, if we have a "javascript project" we should use `TypeScript` to write type annotations

Comment (lucacavallaro) [reply radar/languages-and-frameworks/typescript.md:20]: no ok, you convinced me with these use cases

File: radar/languages-and-frameworks/typescript.md
Comment (balanza) [reply radar/languages-and-frameworks/typescript.md:13]: Sorry my bad, I thought you were talking about "using type annotations on JavaScript" for better IDE DX. 

So I don't understand the comment

Comment (lucacavallaro) [reply radar/languages-and-frameworks/typescript.md:13]: No sorry, I wasn't clear

I was saying: an added benefit of TypeScript is that thanks to type annotations not only can you immediately notice errors, but you also have better code autocompletion

Comment (balanza) [reply radar/languages-and-frameworks/typescript.md:13]: ahhh ok, I see. Good, I'll add the benefit

**Pull request merged**
